core/build.gradle

```diff
@@ -21,13 +21,13 @@ dependencies {
     compile "org.apache.lucene:lucene-suggest:$luceneVersion"
 
     // Scopt
-    compile "com.github.scopt:scopt_$scalaVersion:$scoptVersion"
+    compile "com.github.scopt:scopt_%%:$scoptVersion"
 
     // Zip util
     compile 'org.zeroturnaround:zt-zip:1.14'
 
     // XGBoost
-    compile ("ml.dmlc:xgboost4j-spark:$xgboostVersion") { exclude group: 'com.esotericsoftware.kryo', module: 'kryo' }
+    compile ("ml.dmlc:xgboost4j-spark_%%:$xgboostVersion") { exclude group: 'com.esotericsoftware.kryo', module: 'kryo' }
```

Comment:
I think we need both `xgboost4j` and `xgboost4j-spark` here, [according to this](https://github.com/dmlc/xgboost/tree/master/jvm-packages#add-maven-dependency). However, can we also use the gpu-enabled version of xgboost now ? The artifact names are `xgboost4j-gpu_2.12` and `xgboost4j-spark-gpu_2.12`

Reply:
`xgboost4j` already gets pulled in, but let me add it to be explicit.

W.r.t. GPUs: let's leave that baby alone for now and get this upgrade done first. 😅

Reply:
When you finish this PR you can aim at an e2e on GPU throughout data [transformations](https://github.com/NVIDIA/spark-rapids) and [Xgboost](https://github.com/NVIDIA/spark-xgboost-examples) :sunglasses:

Reply:
yes @gerashegalov i was trying to get people to use rapids ;)

Reply:
On the Spark side we require 3.x , so you are on the right track @TuanNguyen27